### ISSUE TITLE
DS4 Rumble through DInput

### ISSUE THREAD

**Issue by Polymega** [Ryochan7/DS4Windows#698]
Hi @Ryochan7 , 

Thanks for keeping the DS4Windows dream alive and well! I wanted to pick your brain about DS4's capabilities as a DInput device, if that's alright? 

The DS4 controller can act as a simple plug-and-play DInput gamepad, as you know. But rumble doesn't seem to work if it's used as a DInput device. A cursory Google search shows people having no rumble support for games like Rocket League and FFXIV using the controller "out of the box" for PC games.

Long story short: Has it ever been determined/figured out why rumble doesn't work out of the box for the DS4? All other DInput functionalities seem to work, so I always found that odd...

**Comment by Ryochan7** [comment]
A lot of games supporting DirectInput just don't support force feedback through DirectInput. Rocket League is one that I can test that I know does not work. However, force feedback works through the virtual DS4 in HTM5 Gamepad Tester so rumble support does work when supported.

https://html5gamepad.com/

**Comment by Polymega** [comment]
Interesting. So Pysonix (Rocket League devs) just didn't bother to implement code for DInput vibration. And yes, I became more confused once i received vibration from HTML5 Gamepad Tester some months back, too. 

I was primarily asking as we recently restored vibration for Silent Hill 2 PC that benefits both DInpit and XInput gamepads (even though the patch name says "XInputVibration"):

https://github.com/elishacloud/Silent-Hill-2-Enhancements/blob/master/Patches/XInputVibration.cpp

Vibration is restored for Xbox controllers and generic DInput-based PC gamepads... except DS4. Complete head-scratcher on our part. 

Thanks for the response!
